src/core/public/doc_links/doc_links_service.ts

```diff
@@ -39,10 +40,28 @@ interface StartDeps {
 export class DocLinksService {
   public setup() {}
   public start({ injectedMetadata }: StartDeps): DocLinksStart {
-    const DOC_LINK_VERSION =
-      injectedMetadata.getOpenSearchDashboardsBranch() === 'main'
-        ? 'latest'
-        : injectedMetadata.getOpenSearchDashboardsBranch();
+    const buildVersion = injectedMetadata.getOpenSearchDashboardsVersion();
+    const pkgBranch = injectedMetadata.getOpenSearchDashboardsBranch();
+    /**
+     * OpenSearch server uses the `branch` property from `package.json` to
+     * build links to the documentation. If set to `main`, it would use `/latest`
+     * and if not, it would use the `version` to construct URLs.
+     */
+    let branch = pkgBranch;
+    if (pkgBranch === 'main') {
+      branch = 'latest';
+    } else {
+      const validDocPathsPattern = /^\d+\.\d+$/;
+      const parsedBuildVersion = parse(buildVersion);
+      if (validDocPathsPattern.test(pkgBranch)) {
+        branch = pkgBranch;
+      } else if (parsedBuildVersion) {
+        branch = `${parsedBuildVersion.major}.${parsedBuildVersion.minor}`;
+      } else {
+        branch = pkgBranch;
```

Comment:
Can't we simplify line 54 to line 62 like this? since we already set `let branch = pkgBranch;` outside the block and line 57 and line 61 are doing the duplicate thing
```
      const validDocPathsPattern = /^\d+\.\d+$/;
      const parsedBuildVersion = parse(buildVersion);
      
      if (!validDocPathsPattern.test(pkgBranch) &&& parsedBuildVersion) {
        branch = `${parsedBuildVersion.major}.${parsedBuildVersion.minor}`;
      } 
```

Reply:
agree and done.